addons/l10n_in/models/account_invoice.py

```diff
@@ -78,6 +190,118 @@ def _onchange_name_warning(self):
+        def get_company_ids(company):
+            company_ids = company
+            if company.parent_id:
+                company_ids |= company.parent_id
+                if company.parent_id.child_ids:
+                    company_ids |= company.parent_id.mapped('child_ids')
+            if company.child_ids:
+                company_ids |= company.mapped('child_ids')
+            return company_ids
+
+        def get_move_lines_by_partner(partner_domain, start_date, end_date, account_ids, period):
+            company_ids = get_company_ids(self.company_id)
+            partner_ids = self.env['res.partner'].search(partner_domain)
+            query = """
+                SELECT line.id
+                FROM account_move_line line
+                JOIN account_move move ON line.move_id = move.id
+                JOIN account_account account ON line.account_id = account.id
+                WHERE move.date >= %s
+                    AND move.date <= %s
+                    AND move.company_id IN %s
+                    AND account.id IN %s
+                    AND account.l10n_in_tds_tcs_section IS NOT NULL
+                    AND account.l10n_in_tds_tcs_section IN (
+                        SELECT id
+                        FROM account_tax_group
+                        WHERE l10n_in_aggregate_period = %s
+                    )
+            """
+            params = [start_date, end_date, tuple(company_ids.ids), tuple(account_ids.ids), period]
+            if partner_ids:
+                query += " AND line.partner_id IN %s"
+                params.append(tuple(partner_ids.ids))
+            self.env.cr.execute(query, params)
+            line_ids = [row[0] for row in self.env.cr.fetchall()]
+            invoice_lines = self.env['account.move.line'].browse(line_ids)
+
+            lines_group_by_partners = {}
+            for line in invoice_lines:
+                partner_identifier = str(line.partner_id.l10n_in_pan or line.partner_id.id)
+                lines_group_by_partners.setdefault(partner_identifier, line)
+                lines_group_by_partners[partner_identifier] |= line
+            return lines_group_by_partners
+
+        def get_tax_totals(start_date, end_date, period=None):
+            lines = get_move_lines_by_partner(
+                partner_domain=partner_domain,
+                start_date=start_date,
+                end_date=end_date,
+                account_ids=account_ids,
+                period=period or 'fiscal_year'
+            )
+            return self._l10n_in_calculate_tax_aggregate_totals(lines)
+
+        partner_domain = [('id', 'in', partners.ids)]
+        if partner_pans := [p.l10n_in_pan for p in partners if p.l10n_in_pan]:
+            # Search with PAN as there might be multiple partner with same PAN and we want to consider that all the records
+            partner_domain = ['|', ('l10n_in_pan', 'in', partner_pans), ('id', 'in', partners.ids)]
```

Comment:
```suggestion
            partners += self.env['res.partner'].search([('l10n_in_pan', 'in', partner_pans), ('id', 'not in', partners.ids)])
```
Well it could be much better if you could just adjust it in a sub query rather than making a search because searching it could just make the process slow :thinking: